src/math/math.hpp

```diff
@@ -18,5 +18,18 @@ inline int sign(double number)
   return 0;
 }
 
+/// Computes the power of an integer (x^iexp) using recursion, which is much faster than std::pow(x, iexp)
+template <int iexp, typename T>
+inline constexpr T pow_int(const T x)
+{
+  static_assert(iexp >= 0, "Exponent must be an integer greater zero.");
+
+  if (iexp == 0)
+    return static_cast<T>(1.);
+  else
```

Comment:
It would be interesting to see how this performs when treating iexp=0 separately.

Could be faster as it saves 1 level of recursion, but could also hinder the tail recursion optimisation.

Reply:
I just tried and both are quite close to each other. I ran it for several times and sometimes the one or the other is (slightly) faster. Considering that the additional treatment of `iexp=0` leads to more code, I would propose to go for the more compact version here?!